baseplate/server/__init__.py

```diff
@@ -296,13 +275,9 @@ def load_app_and_run_server() -> None:
     if einhorn.is_worker():
         einhorn.ack_startup()
 
-    try:
-        # pylint: disable=cyclic-import
-        from baseplate.server.prometheus import start_prometheus_exporter
-    except ImportError:
-        logger.debug("Prometheus exporter not available. pip install prometheus-client to enable.")
-    else:
-        start_prometheus_exporter()
+    from baseplate.server.prometheus import start_prometheus_exporter
+
+    start_prometheus_exporter()
```

Comment:
I'm not totally sold on requiring prometheus on everything that uses baseplate libs. Was this discussed somewhere?

Reply:
Since the company is migrating to Prometheus from Wavefront, we want to add the same functionality that there is for the statsd exporter that ships metrics to Wavefront.  the baseplate.spec documents what the baseplate frameworks should implement and there is a section for Prometheus metrics: https://github.snooguts.net/reddit/baseplate.spec/blob/master/component-apis/prom-metrics.md

Reply:
[discussed offline] Lets swap this back to the try block

Reply:
@nsheaps A follow up from our previous discussions about replacing this `try/except` block with checking the config for metrics being enabled:

It turns out there are two different ways to enable the current statsd metrics (and you cannot set both at the same time):  
1. `metrics.tagging` enables the tagged metrics (described [here](https://github.snooguts.net/reddit/baseplate.spec/blob/master/component-apis/metrics-tagged.md))
2. `metrics.namespace` enables non-tagged metrics (described [here](https://github.snooguts.net/reddit/baseplate.spec/blob/master/component-apis/metrics.md)).

Searching in sourcegraph, it looks like there are approx. 82 repos that set `metrics.namespace` and 20 repos that set `metric.tagging`.  For reference, here are the sourcegraph queries I ran:
```
repo:reddit-service-* repo:contains.file(requirements.txt) file:ini metrics.tagging select:repo
repo:reddit-service-* repo:contains.file(requirements.txt) file:ini metrics.namespace select:repo
```

Since the goal is to enable Prometheus metrics alongside the current statsd metrics (until wavefront is turned off), then we need to enable Prometheus when either `metrics.tagging` or `metrics.namespace` are set.

Another consideration is to create a new config option specifically for Prometheus.  It would be nice to have a separate config for the Prometheus metrics to keep it untangled from the current metric configs options, however there wouldn't be a way to turn on Prometheus automatically without updating the `.ini` file which would make upgrades not backwards compatible.  But I'm open to a discussion either way.